client/state/memberships/coupon-list/test/actions.js

```diff
@@ -34,4 +61,193 @@ describe( 'actions', () => {
 			} );
 		} );
 	} );
+
+	describe( 'requestAddCoupon()', () => {
+		test( 'should dispatch add coupon failure and error notice on failure', async () => {
+			const siteId = 1;
+			const coupon = mockCoupon;
+			nock( 'https://public-api.wordpress.com' )
+				.post( '/wpcom/v2/sites/1/memberships/coupons' )
+				.replyWithError( { code: 'validation_error', message: 'Coupon code is already used' } );
+			const noticeText = 'Added coupon';
+			const dispatchedActions = [];
+
+			const dispatch = ( obj ) => dispatchedActions.push( obj );
+			await requestAddCoupon( siteId, coupon, noticeText )( dispatch );
+
+			const [ addAction, addFailureAction, noticeCreateAction ] = dispatchedActions;
+			expect( addAction ).toHaveProperty( 'coupon' );
+			expect( addAction ).toHaveProperty( 'siteId' );
+			expect( addAction ).toHaveProperty( 'type' );
+			expect( addFailureAction ).toHaveProperty( 'error' );
+			expect( addFailureAction ).toHaveProperty( 'siteId' );
+			expect( addFailureAction ).toHaveProperty( 'type' );
+			expect( noticeCreateAction ).toHaveProperty( 'notice' );
+			expect( noticeCreateAction ).toHaveProperty( 'type' );
```

Comment:
Should we also check for the values associated with these properties? The same would apply to other unit tests using the `toHaveProperty` assertion.